tests/Validation.ContentScan.tests/ContentScanValidatorFacts.cs

```diff
@@ -0,0 +1,256 @@
+            var request = new ValidationRequest(Guid.NewGuid(), new Uri("https://example.com/package.nupkg"));
+            var status = new ValidatorStatus
+            {
+                State = validationStatus,
+                NupkgUrl = null
+            };
+
+            _validatorStateServiceMock
+                .Setup(vss => vss.GetStatusAsync(request))
+                .ReturnsAsync(status);
+
+            var result = await _target.GetResponseAsync(request);
+
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(request), Times.Once);
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(It.IsAny<ValidationRequest>()), Times.Once);
+           
+            Assert.Equal(status.State, result.Status);
+        }
+
+        [Theory]
+        [InlineData(ValidationStatus.Incomplete)]
+        [InlineData(ValidationStatus.NotStarted)]
+        [InlineData(ValidationStatus.Succeeded)]
+        public async Task DoesNotReturnPackageValidationResultForNonTerminalAndSuccessState(ValidationStatus validationStatus)
+        {
+            var request = new ValidationRequest(Guid.NewGuid(), new Uri("https://example.com/package.nupkg"));
+            var status = new ValidatorStatus
+            {
+                State = validationStatus,
+                NupkgUrl = null
+            };
+
+            _validatorStateServiceMock
+                .Setup(vss => vss.GetStatusAsync(request))
+                .ReturnsAsync(status);
+
+            var result = await _target.GetResponseAsync(request);
+
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(request), Times.Once);
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(It.IsAny<ValidationRequest>()), Times.Once);
+
+            Assert.Equal(status.State, result.Status);
+            Assert.Equal(0,result.Results.Count);
+        }
+
+        [Theory]
+        [InlineData(ValidationStatus.Failed)]
+        public async Task DoesReturnPackageValidationResultForFailedState(ValidationStatus validationStatus)
+        {
+            var request = new ValidationRequest(Guid.NewGuid(), new Uri("https://example.com/package.nupkg"));
+            var status = new ValidatorStatus
+            {
+                State = validationStatus,
+                NupkgUrl = null
+            };
+
+            _validatorStateServiceMock
+                .Setup(vss => vss.GetStatusAsync(request))
+                .ReturnsAsync(status);
+
+            var result = await _target.GetResponseAsync(request);
+
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(request), Times.Once);
+            _validatorStateServiceMock
+                .Verify(vss => vss.GetStatusAsync(It.IsAny<ValidationRequest>()), Times.Once);
+
+            Assert.Equal(status.State, result.Status);
+            Assert.True(result.Results.Count > 0);
```

Comment:
I think you can use `NotEmpty` here:

```suggestion
            Assert.NotEmpty(result.Results);
```